docs/proposal/adsFlush.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Kmesh bpf map refresh optimization
+authors:
+- "@LiZhenCheng9527"
+reviewers:
+approvers:
+
+creation-date: 2024-04-28
+
+---
+
+## Backgroud
+
+Because Kmesh provides a new data plane architecture, the control plane still uses istiod. So still need `ads`` for Kmesh and Istiod synchronisation.
+
+## Kmesh xDS Mode Function Design
+
+The process of synchronising Kmesh with istiod is the same as synchronising envoy with istiod:
+
+- **Create grpc links with istiod.**
+  After creating the grpc link, use the `streamAggregatedResources`` method to establish a link with istiod.
+- **Send an initial CDS subscription request to istiod.**
+  After establishing a link, Kmesh sends an initial CDS subscription request to for control plane clustering information.
+- **Processing DiscoveryResponse response messages received from istiod.**
+  In general, to avoid traffic drop, sequencing of updates should follow a make before break model, wherein:
+  - CDS updates (if any) must always be pushed first.
+  - EDS updates (if any) must arrive after CDS updates for the respective clusters. LDS updates must arrive after corresponding CDS/EDS updates.
+  - LDS updates must arrive after corresponding CDS/EDS updates.
+  - RDS updates related to the newly added listeners must arrive after CDS/EDS/LDS updates.
+- **Updating response and generating ack.**
+  And for each xds processed, the xDS.cache and bpf map are both updated.
+- **Send ack to istiod.**
+  A new req or ack is generated when the response from istiod is processed and finally sent to the istiod.
+- **Cycle 4 to 6 steps.**
+
+Kmesh is synchronised with the istiod control surfaces through the six steps described above. However, there are still areas in which it can be optimised.
+
+## Optimisation of xds response handling
+
+- Asynchronously update `bpf map` and `xDS.cache`:
+  At this stage the bpf.map update takes much longer to process than the xDS.cache.
+  The serial processing of updating xDS.cache and then updating the bpf map would greatly reduce the efficiency of Kmesh.
+  So we should use different threads to update bpf.map and xDS.cache.
```

Comment:
To be more precise, it is only to update bpf map asynchrounously, while xds.Cache update is still synchronous